### PR TITLE
HOSTEDCP-1462: Enable TestNodePool and CreateCluster tests on Azure

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
**What this PR does / why we need it**: getting TestCreateCluster and TestNodePool to work with Azure so we can use them as presubmits to get started with.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes # [HOSTEDCP-1462](https://issues.redhat.com/browse/HOSTEDCP-1462)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: test/e2e/util/util.go [hunk 1]
```diff
@@ -1276,7 +1278,7 @@ func EnsurePodsWithEmptyDirPVsHaveSafeToEvictAnnotations(t *testing.T, ctx conte
 			for _, volume := range pod.Spec.Volumes {
 				// Check the pod's volumes, if they are emptyDir or hostPath,
 				// they should include that volume in the annotation
-				if volume.EmptyDir != nil || volume.HostPath != nil {
+				if (volume.EmptyDir != nil && volume.EmptyDir.Medium != corev1.StorageMediumMemory) || volume.HostPath != nil {
```
Comment (bryan-cox) [inline test/e2e/util/util.go:1281]: We need to do this for AWS as well?

Comment (Patryk-Stefanski) [reply test/e2e/util/util.go:1281]: yep this is a general check for local memory , see [here](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/FAQ.md#what-types-of-pods-can-prevent-ca-from-removing-a-node)

**Pull request merged**